src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1826,11 +1805,22 @@ fn resolution_failure(
                             diag.note(&note);
                         }
 
-                        // If the link has `::` in it, assume it was meant to be an intra-doc link.
-                        // Otherwise, the `[]` might be unrelated.
-                        // FIXME: don't show this for autolinks (`<>`), `()` style links, or reference links
```

Comment:
The FIXME is no longer relevant after https://github.com/rust-lang/rust/pull/96187 as I understand, so I removed it.

Reply:
Indeed. Thanks for the cleanup!